primitives/src/authorization.rs

```diff
@@ -0,0 +1,41 @@
+use crate::identity_id::IdentityId;
+use codec::{Decode, Encode};
+use rstd::prelude::Vec;
+
+/// Authorization data for two step prcoesses.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, PartialOrd, Ord)]
+pub enum AuthorizationData {
+    /// Authorization to transfer a ticker
+    TransferTicker(Vec<u8>),
+    /// Any other authorization
+    Custom(Vec<u8>),
+    /// No authorization data
+    None,
+}
+
+impl Default for AuthorizationData {
+    fn default() -> Self {
+        AuthorizationData::None
+    }
+}
+
+/// Authorization struct
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug)]
+pub struct Authorization<U> {
+    /// Enum that contains authorization type and data
+    pub authorization_data: AuthorizationData,
+
+    /// Identity of the organization/individual that added this authorization
+    pub authorized_by: IdentityId,
+
+    /// time when this authorization expires. optional.
+    pub expiry: Option<U>,
+
+    // Extra data to allow iterating over the authorizations.
```

Comment:
Missing third `/`.

Reply:
This was intentional. It's an overarching code comment that applies to the next two elements. Doc comments only apply to the next element. It will not get mentioned when we generate cargo docs but I think that's fine. It's more of a code comment.